File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectBasicsJava/app/src/main/java/pl/edu/uwr/pum/interfaceinjectbasicsjava/computer/Computer.java
package pl.edu.uwr.pum.interfaceinjectbasicsjava.computer;

import javax.inject.Inject;

import pl.edu.uwr.pum.interfaceinjectbasicsjava.computer.component.CPU;

public class Computer {
    private CPU cpu;

    @Inject
    public Computer(CPU cpu) {
        this.cpu = cpu;
    }

    public String cpuName(){
        return cpu.name();
    }
}

File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectiBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/interfaceinjectibasicskotlin/computer/Computer.kt
package pl.edu.uwr.pum.interfaceinjectibasicskotlin.computer

import pl.edu.uwr.pum.interfaceinjectibasicskotlin.computer.components.CPU
import javax.inject.Inject

class Computer @Inject constructor(val cpu: CPU) {
    fun cpuName(): String{
        return cpu.name()
    }
}